deepchecks/recommender/ranking.py

```diff
@@ -7,705 +7,423 @@
+                          k: int = 10) -> float:
     """
-    Calculate the mean average recall at k (MAR@K) for a list of recommendations.
-    Each recommendation should be paired with a list of relevant items. First recommendation list is
-    evaluated against the first list of relevant items, and so on.
+    Computes the mean average F1-score at k for a list of recommendations and their relevant items.
 
     Args:
-        relevant_items (array-like): An M x R list where M is the number of recommendation lists,
-                                     and R is the number of relevant items.
-        recommendations (array-like):  An M x N list where M is the number of recommendation lists and
-                                N is the number of recommended items.
-        k (int): the number of items considered in the recommendation.
+        relevant_items (Union[List[list[int]], np.array]): A list of lists of relevant items.
+        recommendations (Union[List[list[int]], np.array]): A list of lists of recommended items.
+        k (int): The top-k recommendations to consider. Default is 10.
+
     Returns:
-        MAR@K (float): The mean average recall @ k across the recommendations.
+        float: The mean average F1-score at k.
     """
-    ars = []
-    for items, recommendation in zip(relevant_items, recommendations):
-        ar = average_recall_at_k(items, recommendation, k)
-        ars.append(ar)
+    f1_list = []
+    for i,items in enumerate(relevant_items):
+        recommendation = recommendations[i]
+        f1_i = f1_k(items, recommendation, k)
+        f1_list.append(f1_i)
 
-    return np.mean(ars)
+    return np.mean(f1_list)
 
 
-def hits_at_k(relevant_items: np.array, recommendation: np.array, k=10):
+def reciprocal_rank(relevant_item,recommendation: Union[List, np.array, None]) -> float:
     """
-    Calculate the number of hits of relevant items in a ranked list HITS@K.
+    Calculates the reciprocal rank for a given recommendation.
 
     Args:
-        relevant_items (array-like): An 1 x N array of relevant items.
-        recommendation (array-like): An 1 x N array of predicted arrays
-        k (int): the number of items considered in the predicted list
+    - relevant_item: integer representing the relevant item
+    - recommendation: list or numpy array of recommended items
+
     Returns:
-        HITS@K (float):  The number of relevant items in the first k items of a prediction.
+    - float: reciprocal rank of the relevant item in the recommendation
     """
-    if len(recommendation) > k:
-        recommendation = recommendation[:k]
-
-    hits = 0.0
-    for i, item in enumerate(recommendation):
-        if item in relevant_items and item not in recommendation[:i]:
-            hits += 1.0
-
-    if len(recommendation) == 0:
+    if relevant_item is None or len(recommendation) == 0:
         return 0.0
-    return hits / len(recommendation)
+    for rank, rec_item in enumerate(recommendation):
```

Comment:
So in this metric we assume relevant_items (per a specific session) is only one item, while in the previous ones there could be multiple relevant items per session?

Reply:
Reciprocal rank is a way to evaluate how good a recommendation is, but it only looks at the first relevant item in the recommendation list. So, if you're looking for an item on Python Programming and the first book recommended is exactly what you're looking for, reciprocal rank would give a high score. But if the second or third book is also good, reciprocal rank wouldn't take that into account.

This is why there are other evaluation metrics, like DCG and NDCG, that take into account all the relevant items in the recommendation list. These metrics  are the way to go because it gives a more complete picture of how well a recommender system is performing, especially if there are multiple relevant items in the list.

Reciprocal rank is useful when the user is looking for something extremely specific, such as when using search engine.